src/org/thoughtcrime/securesms/ConversationActivity.java

```diff
@@ -1499,4 +1537,82 @@ public void handleEvent(int eventId, Object data1, Object data2) {
       invalidateOptionsMenu();
     }
   }
+
+
+  // in-chat search
```

Comment:
Actually this whole file is getting pretty large (it is very large already), maybe all the search-related things could go into a new class `InChatSearchManager`

Reply:
yip, the file was already large before, however, splitting search, not sure if this makes sense, in general, i do not think, this section adds much complexity.